frontend/README.md

```diff
@@ -0,0 +1,83 @@
+# Frontend application
+
+The project uses a [Node.js (v8)](https://nodejs.org/en/) runtime environment and [React.js](https://reactjs.org/) library for the frontend.
+
+## Module and Library dependencies
+
+The application requires a whole suite of npm modules for building/testing/running the application. All the required modules can be found under package.json file.
+
+### Building
+
+- [webpack](https://webpack.js.org/) : Project building and asset compilation.
+- [babel](https://babeljs.io/) : Adds support for older browsers
+- [sass](https://sass-lang.com/guide) : Clean CSS files
+
+### Testing
+
+- [jest](https://jestjs.io/) : Mocking and writing tests
+- [enzyme](https://github.com/airbnb/enzyme) : Assert, manipulate and traverse react components
+
+### Running
+
+- [react](https://reactjs.org/) : JS library to build single page apps
+- [redux](https://github.com/reduxjs/react-redux) : State management
+- [prop-types](https://www.npmjs.com/package/prop-types) : Runtime object type check
+- [axios](https://github.com/axios/axios) : Library to manage network calls
+
+
+## Directory/Naming convention
+
+The application naming convention is based off of [BEM](http://getbem.com/introduction/) and [ITCSS](https://www.xfive.co/blog/itcss-scalable-maintainable-css-architecture/) naming convention and is structured as follows:
```

Comment:
BEM and ITSCSS are only for SASS/CSS files and filestructure, not the whole app.